integ/src/main.rs

```diff
@@ -0,0 +1,267 @@
+use log::info;
+use snafu::ResultExt;
+use std::env::temp_dir;
+use std::fs;
+use std::{env, process};
+use tokio::time::{sleep, Duration};
+
+use integ::ec2_provider::{create_ec2_instance, terminate_ec2_instance};
+use integ::eks_provider::{get_cluster_info, write_kubeconfig};
+use integ::error::ProviderError;
+use integ::updater::{delete_cluster_resources, label_node, run_brupop};
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+/// The default path for kubeconfig file
+const KUBECONFIG_FILE_NAME: &str = "kubeconfig.yaml";
+
+/// The default region for the cluster.
+const DEFAULT_REGION: &str = "us-west-2";
+const CLUSTER_NAME: &str = "brupop-integration-test";
+
+/// This value configure how long it sleeps between create instance and label instance.
+const INTEGRATION_TEST_DELAY: Duration = Duration::from_secs(60);
+
+/// Stores global arguments.
+#[derive(Debug)]
+struct Args {
+    cluster_name: String,
+    region: String,
+    kube_config_path: String,
+}
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            cluster_name: CLUSTER_NAME.to_string(),
+            region: DEFAULT_REGION.to_string(),
+            kube_config_path: KUBECONFIG_FILE_NAME.to_string(),
+        }
+    }
+}
+
+/// Stores the usage mode specified by the user as a subcommand.
+#[derive(Debug)]
+enum Subcommand {
+    IntegrationTest(IntegrationTestArgs),
+    Clean,
+}
+
+/// Stores user-supplied arguments for the 'integration-test' subcommand.
+#[derive(Debug)]
+struct IntegrationTestArgs {
+    instance_ami_id: String,
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+
+    if let Err(e) = run().await {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+async fn run() -> Result<()> {
+    let (args, subcommand) = parse_args(env::args());
+
+    let cluster_info = get_cluster_info(&args.cluster_name, &args.region)
+        .await
+        .context(error::GetClusterInfo)?;
+
+    let kubeconfig_dir = temp_dir().join(args.kube_config_path);
```

Comment:
nit: I think it's best to perform this computation in the implementation of `default`, in case someone comes along later and wants to provide a path that isn't in `/tmp` at all.